File: app/src/main/java/com/byoutline/kickmaterial/adapters/CategoriesRecyclerViewAdapter.java
package com.byoutline.kickmaterial.adapters;

import android.databinding.ViewDataBinding;
import android.support.annotation.NonNull;
import com.byoutline.kickmaterial.BR;
import me.tatarka.bindingcollectionadapter.BindingRecyclerViewAdapter;
import me.tatarka.bindingcollectionadapter.ItemViewArg;

/**
 * {@link BindingRecyclerViewAdapter} that additionally requires {@link CategoryClickListener}
 *
 * @param <T>
 */
public class CategoriesRecyclerViewAdapter<T> extends BindingRecyclerViewAdapter<T> {
    private final CategoryClickListener categoryClickListener;

    public CategoriesRecyclerViewAdapter(@NonNull ItemViewArg<T> arg, CategoryClickListener categoryClickListener) {
        super(arg);
        this.categoryClickListener = categoryClickListener;
    }

    public void onBindBinding(ViewDataBinding binding, int bindingVariable, int layoutRes, int position, T item) {
        binding.setVariable(BR.categoryClickListener, categoryClickListener);
        super.onBindBinding(binding, bindingVariable, layoutRes, position, item);
    }
}

File: app/src/main/java/com/byoutline/kickmaterial/adapters/CategoriesRecyclerViewAdapter.kt
package com.byoutline.kickmaterial.adapters

import android.databinding.ViewDataBinding
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.byoutline.kickmaterial.BR
import me.tatarka.bindingcollectionadapter2.BindingRecyclerViewAdapter

/**
 * [BindingRecyclerViewAdapter] that additionally requires [CategoryClickListener]
 */
class CategoriesRecyclerViewAdapter<T>(private val categoryClickListener: CategoryClickListener) : BindingRecyclerViewAdapter<T>() {
    override fun onCreateBinding(inflater: LayoutInflater?, layoutId: Int, viewGroup: ViewGroup?): ViewDataBinding {
        val binding = super.onCreateBinding(inflater, layoutId, viewGroup)
        binding.setVariable(BR.categoryClickListener, categoryClickListener)
        return binding
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder?, position: Int, payloads: MutableList<Any>?) {
        super.onBindViewHolder(holder, position, payloads)
    }
}
